operational/vmware/instance_tag_sync/instance_tag_sync.pt

```diff
@@ -0,0 +1,426 @@
+    name $param_cloud_name
+  end
+end
+
+resources "instances", type: "rs_cm.instances" do
+  iterate @clouds
+  cloud_href href(iter_item)
+  filter do
+    state ne: ['inactive', 'terminated']
+  end
+  tags '*'
+end
+
+###############################################################################
+# Datasources
+###############################################################################
+
+datasource "ds_instances_and_tags" do
+   iterate @instances
+   field "href",        href(iter_item)
+   field "id",          val(iter_item,'resource_uid')
+   field "name",        val(iter_item,'name')
+   field "state",       val(iter_item,'state')
+   field "tags",        val(iter_item,'tags')
+   field "type",        "instances"
+end
+
+###############################################################################
+# Scripts
+###############################################################################
+
+
+###############################################################################
+# Policy
+###############################################################################
+
+policy "policy_instance_sync" do
+  validate_each $ds_instances_and_tags do
+    summary_template "{{ rs_project_name }} (Account ID: {{ rs_project_id }}): {{ len data }} instance Names Out of Sync"
+    detail_template <<-EOS
+# instance Name Validation between AWS and Cloud Management
+| instanceId | name | state | tags |
+| -------- | ------ | ------- | -------- |
+{{ range data -}}
+  | {{.id}} | {{.name}} | {{.state}} | {{ range $i, $e := .tags }}{{ if $i }}, {{ end }}{{ $e }}{{ end }} |
+{{ end -}}
+EOS
+    check equals?(1,0)
+    escalate $escalate_resources
+  end
+end
+
+###############################################################################
+# Escalations
+###############################################################################
+
+escalation "escalate_resources" do
+  run "update_tags", data, $param_wstunnel_token, $param_vsphere_username, $param_vsphere_password, $param_tag_category
+end
+
+###############################################################################
+# Cloud Workflow
+###############################################################################
+
+define update_tags($data, $param_wstunnel_token, $param_vsphere_username, $param_vsphere_password, $param_tag_category) do
+  $$debug=false
+  call sys_log('VMWare Instance Tag Sync',to_s($data))
+  call generate_session_token($param_wstunnel_token, $param_vsphere_username, $param_vsphere_password) retrieve $session_token
+  call sys_log("session_token:",$session_token)
+  # $category_id = "urn:vmomi:InventoryServiceCategory:f9c8236f-7320-47f0-8daa-1d7c83a673ab:GLOBAL"
+  call list_categories($param_wstunnel_token, $session_token) retrieve $category_ids
+  $category_id = null
+  foreach $category in $category_ids do
+    call get_category($param_wstunnel_token, $session_token, $category) retrieve $spec
+    if $spec["name"] == $param_tag_category
+      $category_id = $category
```

Comment:
can't you break out of the loop here if you found one?

Reply:
I don't think we have a break function in RCL.